core/src/types/operator/operator.rs

```diff
@@ -1393,28 +1402,36 @@ impl Operator {
     /// # Ok(())
     /// # }
     /// ```
-    pub async fn scan(&self, path: &str) -> Result<Lister> {
+    pub fn scan(&self, path: &str) -> FutureList {
         let path = normalize_path(path);
 
-        if !validate_path(&path, EntryMode::DIR) {
-            return Err(Error::new(
-                ErrorKind::NotADirectory,
-                "the path trying to scan should end with `/`",
-            )
-            .with_operation("list")
-            .with_context("service", self.info().scheme().into_static())
-            .with_context("path", &path));
-        }
+        let fut = FutureList(OperatorFuture::new(
+            self.inner().clone(),
+            path,
+            OpList::default(),
+            |inner, path, _args| {
+                let fut = async move {
+                    if !validate_path(&path, EntryMode::DIR) {
+                        return Err(Error::new(
+                            ErrorKind::NotADirectory,
+                            "the path trying to scan should end with `/`",
+                        )
+                        .with_operation("list")
+                        .with_context("service", inner.info().scheme().into_static())
+                        .with_context("path", &path));
+                    }
 
-        let (_, pager) = self
-            .inner()
-            .list(&path, OpList::new().with_delimiter(""))
-            .await?;
+                    let (_, pager) = inner.list(&path, OpList::new().with_delimiter("")).await?;
```

Comment:
I don't know how this statement is processed.

Reply:
Since the `scan` is just an alias of the `list` with delimiter `""`, you can call `list_with` directly.

Reply:
```rust
        let fut = self.list_with(path).delimiter("");
```

This is enough. It is not necessary to generate a future in `scan` entirely like in the `list`.

Reply:
I have got it